cmake/CMakeLists.txt

```diff
@@ -707,7 +709,9 @@ if (onnxruntime_BUILD_BENCHMARKS)
   if (onnxruntime_PREFER_SYSTEM_LIB)
     find_package(benchmark)
   endif()
-  if (NOT TARGET benchmark::benchmark)
+  if (benchmark_FOUND)
```

Comment:
I think this is a unofficial usage.
The upstream benchmark only provides the CONFIG mode file:
https://github.com/google/benchmark/blob/7d0d9061d83b663ce05d9de5da3d5865a3845b79/src/CMakeLists.txt#L85-L90
And in the Config.cmake.in:
https://github.com/google/benchmark/blob/7d0d9061d83b663ce05d9de5da3d5865a3845b79/cmake/Config.cmake.in
`benchmark_FOUND` is not set.

Reply:
Good point. I tested it only with Conan and it seems that the variable `benchmark_FOUND` is set there.
So we should explicitly set CONFIG mode as well? `find_package(benchmark CONFIG)`

Reply:
> Good point. I tested it only with Conan and it seems that the variable `benchmark_FOUND` is set there. So we should explicitly set CONFIG mode as well? `find_package(benchmark CONFIG)`

And the condition should be `if (TARGET benchmark::benchmark)` since it has a namespace `benchmark`.

Reply:
Fixed in pull request #11155